.ci/cico_rhche_compatibility_test.sh

```diff
@@ -5,129 +5,181 @@
-    PR_EXISTS=0
-    break
+set -e -m +x
+
+function checkPullRequest() {
+  PULL_REQUESTS=$(curl -s https://api.github.com/repos/redhat-developer/rh-che/pulls?state=open | jq '.[].title')
+  PR_EXISTS="false"
+  while read -r pr_title
+  do
+    if [[ "$pr_title" == "\"$RELATED_PR_TITLE\"" ]]; then
+      PR_EXISTS="true"
+      break
+    fi
+  done <<< "$PULL_REQUESTS"
+
+  if [[ "$PR_EXISTS" == "false" ]]; then
+    echo "false"
+  else
+    echo "true"
   fi
-done <<< "$PULL_REQUESTS"
+}
 
-#if PR does not exist, create it
-if [[ $PR_EXISTS -eq 1 ]]; then
+function createPullRequest() {
+  PR_BODY="Tracking changes for fixing compatibility with upstream $CHE_VERSION. This PR was created automatically by Jenkins from job $JOB_NAME"
+  PR_BASE="master"
   echo "Pull request for tracking changes of version $CHE_VERSION was not found - creating new one."
-	
-  #add changes (if there are some) and push branch
+  git push origin "$BRANCH" -f
+  curl -X POST -s -L -H "Content-Type: application/json" -u "${GITHUB_AUTH_STRING}" \
+       --data "{\"title\":\"$RELATED_PR_TITLE\", \"head\":\"$BRANCH\", \"base\":\"$PR_BASE\", \"body\":\"$PR_BODY\"}" \
+       https://api.github.com/repos/redhat-developer/rh-che/pulls
+  return_code=$?
+  if [ ! $return_code -eq 0 ]; then
+    echo $return_code > compatibility_test
+    exit $return_code
+  fi
+}
+
+function runCompatibilityTest() {
+  export USE_CHE_LATEST_SNAPSHOT="true"
+  export BASEDIR=$(pwd)
+  export DEV_CLUSTER_URL=https://devtools-dev.ext.devshift.net:8443/
+  export CHE_VERSION=$(curl -s https://raw.githubusercontent.com/eclipse/che/master/pom.xml | grep "^    <version>.*</version>$" | awk -F'[><]' '{print $3}')
+  if [[ -z $CHE_VERSION ]]; then
+    echo "FAILED to get che version. Finishing script."
+    echo "1" > compatibility_test
+    exit 1
+  fi
+  export RELATED_PR_TITLE="Update to $(echo $CHE_VERSION | cut -d'-' -f 1)"
+  export BRANCH="upstream-check-$CHE_VERSION"
+
+  echo "********** Prepare environment for running compatibility test with upstream version of che: $CHE_VERSION **********"
+
+  eval "$(./env-toolkit load -f jenkins-env.json -r \
+          ^DEVSHIFT_TAG_LEN$ \
+          ^QUAY_ \
+          ^KEYCLOAK \
+          ^BUILD_NUMBER$ \
+          ^JOB_NAME$ \
+          ^ghprbPullId$ \
+          ^RH_CHE \
+          ^GITHUB \
+          ^FABRIC8_HUB_TOKEN)"
+
+  export GITHUB_AUTH_STRING=${GITHUB_PUSH_TOKEN}:${GITHUB_TOKEN_PASSWORD}
+
+  source ./config
+  source ./.ci/functional_tests_utils.sh
+
+  echo "Checking credentials:"
+  checkAllCreds
+  echo "Installing dependencies:"
+  installDependencies
+
+  echo "Configuring remote:"
+  git remote set-url origin https://${GITHUB_AUTH_STRING}@github.com/redhat-developer/rh-che.git
+
+  #Check branch for tracking changes/Create new branch
+  echo "Checking if branch $BRANCH exists."
+  set +e
+  git checkout "$BRANCH" 2>&1 >/dev/null
+  return_code=$?
+  set -e
+  if [ ! $return_code -eq 0 ]; then
```

Comment:
What about `if ! git checkout $BRANCH 2>&1 >/dev/null; then`

Reply:
git can return varying return codes based on the error that happened. I would like to have those error codes printed as the exit code of the script.